demo/csi/digitalocean/plugin.nomad

```diff
@@ -10,16 +10,29 @@ job "digitalocean" {
     task "plugin" {
       driver = "docker"
 
-      config {
-        image = "digitalocean/do-csi-plugin:v2.1.1"
-        args = [
-          "--endpoint=${CSI_ENDPOINT}",
-          "--token=${token}",
-          "--url=https://api.digitalocean.com/",
-        ]
 
+      config {
+        image = "digitalocean/do-csi-plugin:v4.12.0"
+        entrypoint = [ "ash", "/local/run.sh" ]
         privileged = true
       }
+      
+      template {
+        data        = <<EOH
+DO_TOKEN="{{ with nomadVar "secrets/digitalocean_csi_driver" }}{{ .token }}{{ end }}"
+EOH
+        destination = "secrets/do-token.env"
+        env         = true
+      }
+
+      template {
+        data = <<EOF
+#!/usr/bin/env ash
+/bin/do-csi-plugin --endpoint="unix:///csi/csi.sock" --token="$DO_TOKEN" --url="https://api.digitalocean.com/" # optionally you can add parameters such as --volume-limit=20
```

Comment:
At a _very_ quick look at the plugin it looks like it supports taking an environment variable `DIGITALOCEAN_ACCESS_TOKEN` directly rather than the `--token` option, which would be a further improvement. 

Not sure why you removed the reference to `CSI_ENDPOINT`.